packages/mantine/src/components/table/layouts/row-layout/RowLayout.tsx

```diff
@@ -0,0 +1,36 @@
+import {factory, Factory, StylesApiProps, useProps, useStyles} from '@mantine/core';
+import {ReactNode} from 'react';
+import classes from '../../../../styles/RowLayout.module.css';
+import {RowLayoutBody, RowLayoutBodyStylesNames} from './RowLayoutBody';
+import {RowLayoutProvider} from './RowLayoutContext';
+import {RowLayoutHeader, RowLayoutHeaderStyleNames} from './RowLayoutHeader';
+import {RowLayoutIcon, RowLayoutIconStylesNames} from './RowLayoutIcon';
+
+type RowLayoutStylesNames = RowLayoutHeaderStyleNames | RowLayoutIconStylesNames | RowLayoutBodyStylesNames;
+export interface RowLayoutProps extends StylesApiProps<RowLayoutFactory> {
+    children: ReactNode;
+}
+
+export type RowLayoutFactory = Factory<{
+    props: RowLayoutProps;
+    stylesNames: RowLayoutStylesNames;
+    staticComponents: {
+        name: string;
```

Comment:
You changed `name` for `displayName` in the `TableLayout` interface I think?

I wonder if something like this would work

```tsx
staticComponents: TableLayout
```